app/controllers/packages_controller.rb

```diff
@@ -217,7 +217,9 @@ def validate_file_ids(file_ids, current_user, errors)
           errors << "file with id '#{file_id}' could not be found"
         else
           data_file = DataFile.find(id_as_int)
-          if data_file.is_authorised_for_access_by?(current_user)
+          if data_file.is_incomplete_package? || data_file.is_error_file?
```

Comment:
I think you should check if there is a package of the same ID and only do this check if it is infact a package.